tests/test_filters.py

```diff
@@ -183,6 +186,43 @@ def test_value_type(self):
         self.assertEqual(res, (None, 1))
 
 
+class TestValueFromFilter(BaseTest):
+
+    @functional
+    def test_value_from(self):
+        session_factory = self.replay_flight_data('test_value_from')
+
+        ec2 = session_factory().resource('ec2')
+        instance = ec2.create_instances(ImageId='ami-6057e21a', MinCount=1, MaxCount=1)[0].id
+        ec2_client = session_factory().client('ec2')
+        self.addCleanup(ec2_client.terminate_instances, InstanceIds=[instance])
+
+        temp_file = tempfile.NamedTemporaryFile(mode='w')
+        json.dump({"us-east-1": [instance]}, temp_file)
+        temp_file.flush()
+        self.addCleanup(temp_file.close)
+
+        if self.recording: time.sleep(30)
```

Comment:
per pep8 style this should be a separate line rather than a multi-line statement.